MediaSession.cpp

```diff
@@ -543,24 +539,23 @@ CDMi_RESULT MediaKeySession::Decrypt(
     }
     
     DRM_RESULT err = DRM_SUCCESS;
-    if (!initWithLast15) {
-/* PRv3.3 support */
-#ifdef PR_3_3
-      DRM_DWORD rgdwMappings[2];
-      if( f_pcbOpaqueClearContent == NULL || f_ppbOpaqueClearContent == NULL )
-      {
-          dr = DRM_E_INVALIDARG;
-          goto ErrorExit;
-      }
+    DRM_AES_COUNTER_MODE_CONTEXT ctrContext = { 0 };
+    DRM_DWORD rgdwMappings[2];
 
-      *f_pcbOpaqueClearContent = 0;
-      *f_ppbOpaqueClearContent = NULL;
+    if( f_pcbOpaqueClearContent == NULL || f_ppbOpaqueClearContent == NULL )
+    {
+        err = DRM_E_INVALIDARG;
+    }
 
-      ChkBOOL(m_eKeyState == KEY_READY, DRM_E_INVALIDARG);
-      ChkArg(f_pbIV != NULL && f_cbIV == sizeof(DRM_UINT64));
-#else
+    *f_pcbOpaqueClearContent = 0;
+    *f_ppbOpaqueClearContent = NULL;
+
+    ChkBOOL(m_eKeyState == KEY_READY, DRM_E_INVALIDARG);
+    ChkArg(f_pbIV != NULL && f_cbIV == sizeof(DRM_UINT64));
```

Comment:
Wait. Does this compile for you? This uses `goto ExitError` internally and there's no such in the scope.

Reply:
oh yes! `ChkBOOL`, `ChkArg` or `ChkDR` use `DRM_RESULT` variable named as `dr` whereas here `err` is defined. To make a smaller amount of change, it remains as `err`. To use these macros, need a `ExitError` and `DRM_RESULT` to be declared as `dr`

Reply:
There are better ways to do clean error handing. I hate these macros. Please do not use them.